visidata/aggregators.py

```diff
@@ -107,13 +108,48 @@ def _funcRows(col, rows):  # wrap builtins so they can have a .type
 def mean(vals):
     vals = list(vals)
     if vals:
-        return float(sum(vals))/len(vals)
+        if type(vals[0]) is date:
+            vals = [d.timestamp() for d in vals]
+            ans = float(sum(vals))/len(vals)
+            return datetime.date.fromtimestamp(ans)
+        elif isinstance(vals[0], datetime.timedelta):
+            return datetime.timedelta(seconds=vsum(vals)/datetime.timedelta(seconds=len(vals)))
+        else:
+            return float(sum(vals))/len(vals)
 
 def _vsum(vals):
-    return sum(vals, start=type(vals[0] if len(vals) else 0)())  #1996
+    if vals:
+        if type(vals[0]) is date:
+            vd.error('dates cannot be summed')
+            return None
+        return sum(vals, start=type(vals[0])())  #1996
+    else:
+        return 0
 
 # start parameter in sum() added in Python 3.8
 vsum = _vsum if sys.version_info[:2] >= (3, 8) else sum
+def median(vals):
+    if not vals:
+        return None
+    if type(vals[0]) is date:
+        # when the length is even, statistics.median needs to add
+        # two midpoints to average them, so convert to timestamps
+        vals = [d.timestamp() for d in vals]
+        return datetime.date.fromtimestamp(statistics.median(vals))
+    return statistics.median(vals)
+
+def stdev(vals):
+    if vals and len(vals) >= 2:
+        if type(vals[0]) is date:
+            vals = [d.timestamp() for d in vals]
+            return datetime.timedelta(seconds=statistics.stdev(vals))
```

Comment:
This is very interesting, I wonder how we can codify that the output type of stdev for both date and datedelta are datedelta.  It's not quite as simple as making the result type `datedelta` and passing it the float from the generic aggregations, since Python's timedelta (which datedelta is a subclass of) has a constructor with a first parameter of 'days' instead of 'seconds'.